internal/release/cross/release.go

```diff
@@ -62,7 +66,14 @@ func (r *Release) ComputePromotedFile(sourceEnv, targetEnv *v1alpha1.Environment
 	}
 
 	// Only consider promotion if the new merged result is different from existing target
-	if targetRelease == nil || !yml.Compare(targetRelease.File.Yaml, promotedFile.Yaml) {
+	if targetRelease != nil {
```

Comment:
Oh nevermind... I thought we were going to determine if a revision is in sync at the time we need that information. I have a slight preference for colocating this logic to where we need it and keep ComputePromotedFile as simple as possible (IE only computes promotedFile).

If you prefer it like this that's fine.

Reply:
It would have looked more like `ploup` branch where the we check if values are in sync when computing the list, rather than in a precompute stage.